importer_client/python/tools/timesketch_importer.py

```diff
@@ -143,10 +143,30 @@ def upload_file(
         timeline = streamer.timeline
         task_id = streamer.celery_task_id
 
+        analyzer_names = config_dict.get("analyzer_names")
+        timeline_ids = config_dict.get("timeline_ids")
+        if analyzer_names and timeline_ids:
+            streamer.close(analyzer_names=analyzer_names, timeline_ids=timeline_ids)
+        else:
+            streamer.close()
```

Comment:
Instead of using the streamer.close function I would recommend to break the Analyzer execution into a separate function. This allows for a workflow that separates upload and analyzer execution.
This extra function can then be called at the end of the main function when the timeline is confirmed uploaded and if analyzers are provided as arguments.